Terraria/NPC.cs

```diff
@@ -88506,7 +88594,16 @@ public static void CatchNPC(int i, int who = -1)
 			}
 			if (Main.npc[i].catchItem > 0)
 			{
-				int arg_61_0 = Main.npc[i].type;
+				if (Main.npc[i].SpawnedFromStatue)
+				{
+					Vector2 position = Main.npc[i].Center - new Vector2(20f);
+					//Utils.PoofOfSmoke is probably client side magic.
```

Comment:
PoofOfSmoke is creating new dust object. Method is non-existent in our version of TSAPI.